src/tests/stress/pedsim/Ped_util/Global_funs.enc

```diff
@@ -0,0 +1,230 @@
+module Global_funs
+import Agent_passive
+import XML
+fun inv_sqrt(input : int) : real
+  var ret = 0.0
+  EMBED (unit)
+    {
+  float x = #{input};
+  float xhalf = 0.5f * x;
+  int i = *(int*)&x;
+  i = 0x5f3759df - (i >> 1);
+  x = *(float*)&i;
+  #{ret}= x*(1.5f - xhalf*x*x);
+  }
+  END
+  ret
+end
+fun find_extreme(agents : [Agent]) : (int, int, int, int)
+  var extreme = (agents(0).x,agents(0).y,agents(0).x,agents(0).y)
+  var i = 0
+  for a <- agents do
+    i = i + 1
+    extreme = extreme_check(a.pos(), extreme)
+    for index <- [0..|a.targets_x| - 1] do
+      extreme = extreme_check(((a.targets_x)(index), (a.targets_y)(index)), extreme)
+    end
+  end
+  match extreme with
+    case (xmax, ymax, xmin, ymin) =>
+      (xmax, ymax, xmin, ymin)
+    end
+  end
+end
+fun extreme_check(a : (int, int), b : (int, int, int, int)) : (int, int, int, int)
+  match (a, b) with
+    case ((x, y), (xmax, ymax, xmin, ymin)) =>
+      (max(x, xmax), max(y, ymax), min(x, xmin), min(y, ymin))
+    end
+  end
+end
+
+fun copy_ar(arr : [int]) : [int]
```

Comment:
In my experience, attempts at using parametric and/or polymorphic constructs have resulted in a lot of crashes and no benefit. Therefore, I'll not use any parametric method/functions or polymorphism.

This might no longer be a problem but I don't care, I got burned to many times.